cpp/src/parquet/meson.build

```diff
@@ -0,0 +1,283 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+parquet_srcs = files(
+    '../generated/parquet_types.cpp',
+    'arrow/path_internal.cc',
+    'arrow/reader.cc',
+    'arrow/reader_internal.cc',
+    'arrow/schema.cc',
+    'arrow/schema_internal.cc',
+    'arrow/variant_internal.cc',
+    'arrow/writer.cc',
+    'bloom_filter.cc',
+    'bloom_filter_reader.cc',
+    'chunker_internal.cc',
+    'column_reader.cc',
+    'column_scanner.cc',
+    'column_writer.cc',
+    'decoder.cc',
+    'encoder.cc',
+    'encryption/encryption.cc',
+    'encryption/internal_file_decryptor.cc',
+    'encryption/internal_file_encryptor.cc',
+    'exception.cc',
+    'file_reader.cc',
+    'file_writer.cc',
+    'geospatial/statistics.cc',
+    'geospatial/util_internal.cc',
+    'geospatial/util_json_internal.cc',
+    'level_comparison.cc',
+    'level_conversion.cc',
+    'metadata.cc',
+    'page_index.cc',
+    'platform.cc',
+    'printer.cc',
+    'properties.cc',
+    'schema.cc',
+    'size_statistics.cc',
+    'statistics.cc',
+    'stream_reader.cc',
+    'stream_writer.cc',
+    'types.cc',
+    'xxhasher.cc',
+)
+
+thrift_dep = dependency('thrift')
+parquet_deps = [arrow_dep, rapidjson_dep, thrift_dep]
+
+openssl_dep = dependency('openssl', required: needs_parquet_encryption)
+if openssl_dep.found()
```

Comment:
It seems that Parquet encryption is enabled even when `parquet_require_encryption=disabled` and system OpenSSL exists.

I think that the following is expected:
* `parquet_require_encryption=enabled`:
  * `dependency('openssl')` must be found
  * Parquet encryption is built
* `parquet_require_encryption=auto`:
  * Parquet encryption is built only when `dependency('opnessl')` is found
* `parquet_require_encryption=disabled`:
  * Don't execute `dependency('openssl')`
  * Parquet encryption is not built

Reply:
OK I think the latest iteration simplifies this by branching only when the `parquet_require_encryption` option is enabled. 

Note that the `auto` feature in Meson is still either in an enabled or disabled state; it just depends if `auto_features=enabled` or `auto_features=disabled` when the project is configured

Reply:
Oh, really?
Do you know the `auto` feature documentation?

I saw https://mesonbuild.com/Reference-manual_returned_feature.html . It includes `feature.allowed()` and `feature.auto()`. It seems that `get_option('parquet_require_encryption')` may be `auto`.